app/models/miq_server/worker_management/monitor.rb

```diff
@@ -40,21 +40,14 @@ def worker_not_responding(w)
   end
 
   def sync_workers
-    result = {}
-    MiqWorkerType.worker_class_names.each do |class_name|
-      begin
-        c = class_name.constantize
-        raise NameError, "Constant problem: expected: #{class_name}, constantized: #{c.name}" unless c.name == class_name
```

Comment:
Looks like this line was lost in the refactoring.  It was probably added for better diagnostics.

Reply:
Well removing this was actually the point of this change.  We have control over what makes it into `miq_worker_types` so this shouldn't happen.  I believe this was from before we had the `miq_worker_types` table back when it was `self.class.monitor_class_names`

Reply:
thanks for the explanation

Reply:
No, this was added specifically because of that weird issue with Ruby where you can ask for a particular constant and get back the wrong one if someone happened to defined one in your module ancestry.  See the commit message in afe1f02b3eb50b7b840a9c7e8939c11d3a3b2b2f

Reply:
That being said, afe1f02 was added 2019-10-25, and the refactor into the database table was added in 2947aac92c27defc0818e00fa4760af0b4974bab on 20129-11-19

Reply:
Yeah I was thinking if we prevent bad class names from hitting MiqWorkerType in the first place then we don't need to check this at runtime, and everyone else who does constantize on a worker_class_name doesn't have to duplicate this check.

Reply:
@agrare Oh I'm not disagreeing with you ...just clarifying what the purpose of this line actually was and showing that yes, it was from before the refactor to a table

Reply:
Yeah, I agree.  You're both right. 👍

Reply:
✂️ 🔥 ✂️ 🔥 ✂️ 🔥 

🍰  🍪 👏 🙇 😍 🎉